src/sbvr-api/sbvr-utils.ts

```diff
@@ -1024,7 +1024,24 @@ const getIdField = (
 	>,
 ) => getAbstractSqlModel(request).tables[resolveSynonym(request)].idField;
 
-export const getAffectedIds = async ({
+export const getAffectedIds = async (args: {
+	req: HookReq;
+	request: HookRequest;
+	tx: Db.Tx;
+}): Promise<number[]> => {
+	const { request } = args;
+	if (request.affectedIds) {
+		return request.affectedIds;
+	}
+	// We store the affected ids promise so we only have to fetch it once per request
+	if (request._affectedIdsPromise == null) {
+		request._affectedIdsPromise = getAffectedIdsNoCache(args);
+	}
+
+	return request._affectedIdsPromise;
```

Comment:
Can you make this
```suggestion
	if (request.pendingAffectedIds == null) {
		request.pendingAffectedIds = $getAffectedIds(args);
	}

	await request.pendingAffectedIds;
	delete request.pendingAffectedIds;
	
	return request.affectedIds
```
Few things
1. using `pending` for the pending promise as it should be a bit clearer
2. deleting the promise once it's resolved so it can be GC'ed immediately
3. using `$getAffectedIds` for the internal version, mainly because it matches how we name memoized functions generally

Reply:
I thought about point 2 as well, but since it wasn't already in the api I thought there was a reason for this.
Will do :+1: